packages/dialtone-tokens/themes/config.js

```diff
@@ -61,3 +190,341 @@ function _setThemeAttributeOnRoot (theme, brand, rootNode) {
+    );
+  }
+
+  if (typeof brandTheme.brand.name !== 'string' || !brandTheme.brand.name) {
+    throw new TypeError(
+      '[Dialtone] initDialtoneTheme: brandTheme.brand.name must be a non-empty string.',
+    );
+  }
+
+  if (typeof brandTheme.brand.css !== 'string') {
+    throw new TypeError(
+      '[Dialtone] initDialtoneTheme: brandTheme.brand.css must be a string containing CSS.',
+    );
+  }
+
+  // Validation: mode must be valid
+  if (mode !== 'light' && mode !== 'dark') {
+    throw new TypeError(
+      `[Dialtone] initDialtoneTheme: mode must be 'light' or 'dark', got '${mode}'.`,
+    );
+  }
+
+  // Validation: rootNode must be a valid element
+  if (!rootNode || (typeof rootNode !== 'object')) {
+    throw new TypeError(
+      '[Dialtone] initDialtoneTheme: rootNode must be an HTMLElement or host element with shadowRoot.',
+    );
+  }
+
+  // Warn if someone passed shadowRoot directly instead of the host
+  if (rootNode instanceof ShadowRoot) {
+    console.warn(
+      '[Dialtone] You passed a ShadowRoot directly to initDialtoneTheme(). ' +
+      'Please pass the host element instead. The function will access shadowRoot automatically.\n' +
+      'Correct: initDialtoneTheme(brand, mode, hostElement)\n' +
+      'Incorrect: initDialtoneTheme(brand, mode, hostElement.shadowRoot)',
+    );
+  }
+
+  if (rootNode?.shadowRoot) {
+    rootNode = rootNode.shadowRoot;
+  }
+
+  // Check for duplicate initialization (idempotency protection with soft warnings)
+  const existing = initializationState.get(rootNode);
+  if (existing) {
+    if (existing.brand === brandTheme.brand.name && existing.mode === mode) {
+      console.warn(
+        `[Dialtone] Theme already initialized for this rootNode with brand '${brandTheme.brand.name}' and mode '${mode}'. ` +
+        'Re-applying the same theme may be unnecessary. ' +
+        'If you need to switch themes dynamically, use setBrand() or setMode() instead of calling initDialtoneTheme() again.',
+      );
+    } else {
+      console.warn(
+        `[Dialtone] Theme already initialized for this rootNode. ` +
+        `Previous: brand='${existing.brand}', mode='${existing.mode}'. ` +
+        `New: brand='${brandTheme.brand.name}', mode='${mode}'. ` +
+        'Re-initializing with different parameters. Consider using setBrand()/setMode() for dynamic switching.',
+      );
+    }
+  }
+
+  // Warn if this looks like an embedded app using the wrong rootNode
+  if (rootNode === document.documentElement && existing) {
+    const hasThemeStyles = document.querySelector('#dialtone-css-core') !== null;
+    if (hasThemeStyles) {
+      console.warn(
+        '[Dialtone] Multiple calls to initDialtoneTheme() detected on document.documentElement. ' +
+        'If you are in an embedded app/micro-frontend, you may need to pass your own container element instead. ' +
+        'Example: initDialtoneTheme(core, brand, mode, myAppContainerElement)',
+      );
+    }
+  }
```

Comment:
This is the important one. We need to remove the `&& existing` from the condition here. This will prevent an embedded app from checking on the first init (which is exactly what we want).

If hasThemeStyles is true, we want to trigger an error rather than a warning, and then return to exit the function. It's important that we do NOT continue and set a style when this case happens.